### DISCUSSION TITLE
Testing Infinitic workflows

### DISCUSSION THREAD

**Discussion by runenielsen** [infiniticio/infinitic#182] [Q&A] [answered]
Thanks for the excellent tool!

Is it possible to test Infinitic workflows? Something similar to this: https://orkes.io/content/docs/how-tos/sdks/java-sdk/testing_framework

There doesn't seem to be test in the examples. I was also not able to find much about testing in the Infinitic documentation, only this page, that have a section on unit testing predicates: https://docs.infinitic.io/docs/workflows/signals#unit-testing-predicates

I'm more thinking of integration / component testing workflows in general.

Thanks in advance.

**Comment by geomagilles** [comment, marked as answer]
Hi @runenielsen , thanks for the kind words. I realize I should add an entry in the documentation for workflow testing. The module `infinitic-tests` do a bunch of tests on workflows and could be used as examples. It uses an in-memory implementation, so you do not need a pulsar instance to run them.

**Reply by runenielsen** [reply]
Hi Gilles.

Excellent, thank you for the quick answer, that was just what I was looking for.

Best regards,
Rune
